storage/gcloud/aio/storage/storage.py

```diff
@@ -215,9 +213,10 @@ async def download(self, bucket: str, object_name: str, *,
 
     async def download_to_filename(self, bucket: str, object_name: str,
                                    filename: str, **kwargs: Any) -> None:
-        with open(filename, 'wb+') as file_object:
-            file_object.write(await self.download(bucket, object_name,
-                                                  **kwargs))
+        async with aiofiles.open(filename, mode='wb+') as file_object:
```

Comment:
Unfortunately, this will break `gcloud-rest` since `aiofiles` will still be attempted to get used in that (non-async) use-case.

What you probably need to do here is make use of the `if BUILD_GCLOUD_REST` block in a similar way to how we handle `time.sleep()` vs `asyncio.sleep()` and such, eg. with something along the lines of the following:
```python
if BUILD_GCLOUD_REST:
    file_open = open
else:
    from aiofiles import open as file_open

# SNIP
    async def download_to_filename(..):
        async with file_open(..) as f:
```